RBI-15: Create Recommended Meals UI

FarcasDavid opened the pull request:
-Finished UI for recommendations Collection View
-Added Recommendations label and See All button + localization(romanian)
-Renamed some keys for localizations with more specific names

Commit:
no message

Commit:
Merge branch 'develop' of https://github.com/FarcasDavid/RecipeBook into feature/RBI15-UI-Recommended-Meals

Commit:
RBI-15:
-Finished UI for recommendations Collection View
-Added Recommendations label and See All button + localization(romanian)
-Renamed some keys for localizations with more specific names

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -16,8 +16,13 @@ class HomeScreenViewController: UIViewController {
     @IBOutlet private weak var categoriesTitleLabel: UILabel!
     @IBOutlet private weak var seeAllCategoriesButton: UIButton!
+    @IBOutlet private weak var recommendationsCollectionView: UICollectionView!
+    @IBOutlet private weak var seeAllRecommendationsLabel: UIButton!
```
This is a button not a label

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -27,6 +32,7 @@ class HomeScreenViewController: UIViewController {
         setupUI()
         setupCategoriesCollectionView()
         loadAllCategories()
+        setupRecommendationsCollectionView()
```
Just for clarity, leave the setup methods first and the methods for loading the data last.

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -91,18 +99,48 @@ extension HomeScreenViewController {
+
+                return viewModel.categoriesNumberOfItemsInSection
+            }
+            return viewModel.recommendationsNumberOfItemsInSection
```
When we deal with multiple collection views, the best practice is to use an `if` statement for each one.

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -91,18 +99,48 @@ extension HomeScreenViewController {
+        _ collectionView: UICollectionView,
+        numberOfItemsInSection section: Int) -> Int {
+            if collectionView == self.categoriesCollectionView {
+
```
Remove whitespace

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -91,18 +99,48 @@ extension HomeScreenViewController {
+    func collectionView(
+        _ collectionView: UICollectionView,
+        numberOfItemsInSection section: Int) -> Int {
+            if collectionView == self.categoriesCollectionView {
```
Self is not needed.

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -91,18 +99,48 @@ extension HomeScreenViewController {
+            let category = viewModel.categories[indexPath.row]
+            cell.setup(with: category)
+            return cell
+        } else {
```
Same as my previous comment, use an `if` statement for each collection view. Also, `self` is not needed.

edwardchristian97 commented on RecipeBook/ViewControllers/HomeScreenViewController/HomeScreenViewController.swift:
```diff
@@ -113,7 +151,10 @@ extension HomeScreenViewController: UICollectionViewDelegateFlowLayout {
+            if collectionView == self.categoriesCollectionView {
+                return categoriesCellSize
+            }
+            return recommendationsCellSize
```
`if` statements for each collection view and `self` is not needed.

edwardchristian97 commented on RecipeBook/ViewModels/HomeScreenViewModel.swift:
```diff
@@ -20,10 +20,15 @@ class HomeScreenViewModel {
 
+    var recommendationsNumberOfItemsInSection: Int {
+//        return recommendations.count <= 5 ? recommendations.count : 5
+        return 5
```
Use the code you commented, you can never be sure that you have at least 5 meals coming from the server.

Commit:
RBI-15:
- Code refractoring
- Created Meal object in MealModel.swift, an object that will store All Meals from the API locally
